src/utils/logging.hpp

```diff
@@ -22,31 +22,33 @@
 #include <filesystem>
 #include <optional>
 
+#include <fmt/format.h>
 #include <spdlog/fmt/ostr.h>
 #include <spdlog/sinks/stdout_color_sinks.h>
 #include <spdlog/spdlog.h>
 
+#include <boost/preprocessor/comparison/equal.hpp>
+#include <boost/preprocessor/control/if.hpp>
+#include <boost/preprocessor/variadic/size.hpp>
+
 namespace logging {
 
 // TODO (antonio2368): Replace with std::source_location when it's supported by
 // compilers
-template <typename... Args>
-void AssertFailed(const char *file_name, int line_num, const char *expr, const Args &...msg_args) {
-  std::optional<std::string> message;
-  if constexpr (sizeof...(msg_args) > 0) {
-    message.emplace(fmt::format(msg_args...));
-  }
-
+inline void AssertFailed(const char *file_name, int line_num, const char *expr, const std::string &message) {
```

Comment:
This change is necessary to utilize the compile time checks in fmt, right?

Reply:
yes